src/dcc.cpp

```diff
@@ -7,13 +6,15 @@
 
 #include "compiler.h"
 #include "macros.h"
+#include "option.h"
 #include "parser.h"
 #include "tokenizer.h"
 #include "tokens.h"
 
 int main(int argc, char** argv) {
-  if (argc != 2) ERR_AND_EXIT("Usage: " << argv[0] << " <file.dlang>");
-  auto inFile = std::ifstream(argv[1]);
+  if (argc < 2) ERR_AND_EXIT("Usage: " << argv[0] << " (options) <file.dlang>");
+  getOptions(argc, argv);
+  auto inFile = std::ifstream(argv[argc - 1]);
```

Comment:
What about a case like `./dcc main.dlang -o main`?

Reply:
Can the input file name go anywhere or does it have to be either first or last?